common/util/sha256.h

```diff
@@ -0,0 +1,138 @@
+     the following disclaimer.
+
+   - Redistributions in binary form must reproduce the above
+     copyright notice, this list of conditions and the following
+     disclaimer in the documentation and/or other materials provided
+     with the distribution.
+
+   - Neither the name of Internet Society, IETF or IETF Trust, nor
+     the names of specific contributors, may be used to endorse or
+     promote products derived from this software without specific
+     prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+   DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+   NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+   LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+   HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+   CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+   OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+   EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+   Implementation of the SHA256 Secure Hash Algorithm as defined in the U.S.
+   National Institute of Standards and Technology Federal Information Processing
+   Standards Publication (FIPS PUB) 180-3 published in October 2008 and formerly
+   defined in its predecessors, FIPS PUB 180-1 and FIP PUB 180-2. See:
+
+   The algorithm description is available at
+   http://csrc.nist.gov/publications/fips/fips180-3/fips180-3_final.pdf
+
+   The code is derived from: https://www.rfc-editor.org/rfc/rfc6234.txt
+*/
+
+#ifndef VERIBLE_COMMON_UTIL_SHA_H_
+#define VERIBLE_COMMON_UTIL_SHA_H_
+
+#include <array>
+#include <cstdint>
+#include <string>
+
+#include "absl/strings/string_view.h"
+
+namespace verible {
+
+// The size of SHA256 hash in bytes.
+constexpr int32_t kSha256HashSize = 32;
+// The size of a single message block, in bytes.
+constexpr int32_t kSha256MessageBlockSize = 64;
+
+// The context information for the SHA-256 hashing operation.
+class Sha256Context {
+ public:
+  Sha256Context() { Reset(); }
+
+  // Finishes off the digest calculations and returns the 256-bit message
+  // digest. Resets the context in preparation for computing a new SHA-256.
+  // NOTE: The first octet of hash is stored in the element with index 0, the
+  // last octet of hash in the element with index 27/31.
+  std::array<uint8_t, kSha256HashSize> BuildAndReset();
+
+  // Adds an array of octets as the next portion of the message.  Returns false
+  // if the accumulated message is too large (>2 Exabytes). Can be called
+  // multiple times to incrementally build the digest.
+  bool AddInput(absl::string_view message);
+
+  bool IsOverflowed() { return overflowed_; }
```

Comment:
make `const` ?
(maybe also comment when that happens, with 2 Exibytes ?)

Probably for us _2 Exibytes ought to be enough for anybody..._

Reply:
Done